### PR TITLE
Mystic Miscellaneous Matches

### PR DESCRIPTION
**Pull request by JaThePlayer**
Also fixes some types in sprite.c/.h, though there are still tons of weird things going on there.

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀 This PR matches 6 functions (+0.04%) on `us`.

⚠️ This PR introduces a warning:
- src/battle/partner/sushie.c:110: warning: assignment makes integer from pointer without a cast
- src/battle/partner/sushie.c:121: warning: initialization makes integer from pointer without a cast
- src/battle/partner/sushie.c:124: warning: passing arg 1 of `remove_effect' makes pointer from integer without a cast
- src/battle/partner/watt.c:104: warning: passing arg 6 of `fx_bulb_glow' from incompatible pointer type
- src/sprite.c:345: warning: comparison between pointer and integer

File: src/battle/partner/watt.c [hunk 1]
```diff
@@ -11,6 +11,7 @@ extern s32 D_8023C1C8;
 extern EffectInstance* D_8023C1CC;
 extern EffectInstance* D_8023C1D0;
 extern EffectInstance* D_80239A0C_7054FC;
+extern EffectInstance* D_8023C1B0;
```
Comment (ethteck) [inline src/battle/partner/watt.c:14]: Could you move this to be the first extern in the list (so it's in memory order)? B0, B4, etc

File: src/sprite.c [hunk 2]
```diff
@@ -354,7 +392,7 @@ s32 func_802DE748(s32 arg0, s32 arg1) {
         return -1;
     }
 
-    return componentList[arg1]->unk_4F;
+    return (u8)componentList[arg1]->unk_4C;
```
Comment (ethteck) [inline src/sprite.c:395]: why was this changed?

Comment (ethteck) [reply src/sprite.c:395]: actually, maybe it's correct. I'm just wondering if there was something to corroborate it

Comment (ethteck) [reply src/sprite.c:395]: ohhhh it's the ```
if (arg2 != 0) {
                    component->unk_4C |= 0x10000000;
                } else {
                    component->unk_4C &= ~0xF0000000;
                }
                ```
                I think this indicates it's actually a bitfield or something, but we can worry about that later. Though, instead of the u8 casts, we should probably be doing & 0xFF on that field.

**Review by ethteck — CHANGES_REQUESTED** [review]
Great jorb!

File: src/sprite.c (hunk 2 above)
Comment (ethteck) [reply src/sprite.c:395]: (there's another u8 cast or two on that field elsewhere in this PR, so this applies to all of those)

Comment (JaThePlayer) [reply src/sprite.c]: changed to `& 0xFF`

File: src/battle/partner/watt.c [hunk 3]
```diff
@@ -91,7 +92,26 @@ ApiStatus func_8023859C_70408C(Evt* script, s32 isInitialCall) {
...
+        fx_bulb_glow(0, x, y, z, 1.0f, &D_8023C1B0);
+    }
+
+    script->functionTemp[0]--;
+    if (script->functionTemp[0] == 0) {
+        ((BulbGlowFXData*)D_8023C1B0->data)->unk_14 = 5;
+        return ApiStatus_DONE2;
+    }
+
+    return ApiStatus_BLOCK;
```
Comment (nanaian) [inline src/battle/partner/watt.c:113]: I'd prefer this as an else

File: src/sprite.c [hunk 4]
```diff
@@ -226,7 +226,24 @@ void spr_init_component_anim_state(SpriteComponent* comp, s16*** anim) {
     comp->scale.z = 1.0f;
 }
 
+
+// matches, but has a very questionable cast, spr_init_component_anim_state args are probably wrong
```
Comment (nanaian) [inline src/sprite.c:230]: Does this func actually match? If so it might be wise to remove the asm/NON_EQUIVALENT wrapper and just have this as a `// TODO improve match`

Comment (JaThePlayer) [reply src/sprite.c]: yea it does match, no clue why I added the NON_EQUIVALENT thing here tbh lol
